dask/highlevelgraph.py

```diff
@@ -200,17 +225,23 @@ def __init__(
         dependencies: Mapping[str, Set],
         key_dependencies: Optional[Mapping[Hashable, Set]] = None,
     ):
-        self.__keys = None
+        self._keys = None
         self.layers = layers
         self.dependencies = dependencies
         self.key_dependencies = key_dependencies
 
+        # Makes sure that all layers are `Layer`
+        self.layers = {
+            k: v if isinstance(v, Layer) else BasicLayer(v)
+            for k, v in self.layers.items()
+        }
```

Comment:
Ah, I see you've already included this check! I should have looked here before opening up https://github.com/dask/dask/pull/6716 : )

Reply:
Yeah sorry for not mentioning  it earlier but I forgot all about that change :)